tests/interfaces/test_qnode_autograd.py

```diff
@@ -713,6 +713,295 @@ def cost(a, b, c, weights):
             # to the first parameter of circuit1.
             assert circuit1.qtape.trainable_params == {1, 2}
 
+    def test_second_derivative(self, dev_name, diff_method, mocker, tol):
+        """Test second derivative calculation of a scalar valued QNode"""
+        if diff_method not in {"parameter-shift", "backprop"}:
+            pytest.skip("Test only supports parameter-shift or backprop")
+
+        dev = qml.device(dev_name, wires=1)
+
+        @qnode(dev, diff_method=diff_method, interface="autograd")

```

Comment:
I wonder if we should create an "analytic circuit" set of fixtures for the testing suite and be able to easily reuse the quantum function, result, derivative, and hessian across the different interfaces.

Reply:
This is a good point 🤔 many more tests like this exist across the interface test suite, so maybe something worth considering in a new PR?